app/src/main/java/com/sethpthomas91/httpserver/response/Header.java

```diff
@@ -32,24 +33,45 @@ else if (uniformResourceIdentifier.equals("/head_request")) {
         else if (uniformResourceIdentifier.equals("/redirect")) {
             newLocation = "http://127.0.0.1:5000/simple_get";
         }
+        else if (uniformResourceIdentifier.equals("/text_response")) {
+            contentType = "text/plain;charset=utf-8";
+        }
+        else if (uniformResourceIdentifier.equals("/html_response")) {
+            contentType = "text/html;charset=utf-8";
+        }
+        else if (uniformResourceIdentifier.equals("/json_response")) {
+            contentType = "application/json;charset=utf-8";
+        }
+        else if (uniformResourceIdentifier.equals("/xml_response")) {
+            contentType = "application/xml;charset=utf-8";
+        }
```

Comment:
Same here, feels like a Header could be told what its content type is rather than being conditional internally.

Reply:
Fixed in the newest branch. I moved all of the logic from the header/body/status line except for setting the header for content length.